build.sbt

```diff
@@ -44,7 +44,9 @@ val localAndCloudCommonDependencies = Seq(
     "net.snowflake" % "spark-snowflake_2.12" % "2.10.0-spark_3.2",
     "org.apache.commons" % "commons-lang3" % "3.12.0",
     "org.xerial" % "sqlite-jdbc" % "3.36.0.3",
-    "com.github.changvvb" %% "jackson-module-caseclass" % "1.1.1"
+    "com.github.changvvb" %% "jackson-module-caseclass" % "1.1.1",
+    "com.azure.cosmos.spark" % "azure-cosmos-spark_3-1_2-12" % "4.11.1",
```

Comment:
Can we use `com.azure.cosmos.spark:azure-cosmos-spark_3-2_2-12:4.11.2` if possible, assuming this cosmosdb 3.2 package can work well with Spark 3.1?

Reply:
Unfortunately no. Spark 3.1 and 3.2 are not that compatible.

I'm thinking of an improvement to add the jar dependency from the client side, so we can remove this dep from the core.
But then the client needs the knowledge of the cluster runtime version before submitting the job, I'm still trying to figure out a clear way to do that.
Or we can create 2 different Feathr JARs, one for Spark 3.1 and the other for 3.2, that's also the standard way that other Spark-related projects followed.

Reply:
interesting... in that case, maybe we can use Maven for those jars?

i.e. don't package them in the fat Feathr jar, but when submitting the job, always use Maven to get  `com.azure.cosmos.spark`, and as you said, in the client side users can specify the spark version?

Reply:
Both DataBricks and Synapse support specifying Spark version, but in very different ways. In DataBricks user can set the DataBricks Runtime version in the job submission request if `new_cluster` is used, but in Synapse the Spark version is set when creating the cluster, and cannot be changed later.
If we want the client to set JAR deps, it may need a solid way to detect all Spark cluster info and compose the dep list from it, but I'm afraid this procedure could be pretty error-prone.